load-tests/smtp_client/src/service.bal

```diff
@@ -76,3 +84,18 @@ isolated function startPerfTest(decimal duration) returns error? {
     }
     return;
 }
+
+listener email:ImapListener imapListener = check new ({
+    host: EMAIL_SERVER_HOST,
+    port: IMAP_PORT,
+    username: IMAP_CLIENT_USER,
+    password: IMAP_CLIENT_PASSWORD,
+    pollingInterval: 30,
+    security: email:START_TLS_NEVER
+});
+
+isolated service on imapListener {
+    isolated remote function onMessage(email:Message email) {
+        resultCounter.incrementReceivedCount();
```

Comment:
There is only one email publisher and if the received content is erroneous this remote method will not be invoked. So, I do not think we need to validate it here.

Reply:
I think we should set `completed = true;` in here. Otherwise, the test will complete even if the listener takes some time to receive msgs,

Reply:
This remote method is invoked whenever an email is received. Hence this is not the perfect place to implement this logic. What we can do is after the content publish is completed we could wait for sometime before updating the state. This might not be 100% accurate because there will be lost messages. But IMHO this is the best way to do it.

Check this commit: [82db6fc97037d821d43c6290bc4a8eaaff6e190d](https://github.com/ballerina-platform/module-ballerina-email/pull/445/commits/82db6fc97037d821d43c6290bc4a8eaaff6e190d)

Reply:
In kafka, once the test time is elapsed, we send a msg with a different kind of content. In the receiving end, we analyze the msg and if we find the finished content, we set `completed = true;`. Can't we do that here?

Reply:
Yes I think this is good. Check the commit: [5925b6e33d6a42e46c2e7eb07a5a38288660440a](https://github.com/ballerina-platform/module-ballerina-email/pull/445/commits/5925b6e33d6a42e46c2e7eb07a5a38288660440a)